bpf/lib/nodeport.h

```diff
@@ -2907,6 +2914,22 @@ static __always_inline int nodeport_lb4(struct __ctx_buff *ctx,
 	if (backend_local || !nodeport_uses_dsr4(&tuple)) {
 		struct ct_state ct_state = {};
 
+#if !(defined(ENABLE_CLUSTER_AWARE_ADDRESSING) && defined(ENABLE_INTER_CLUSTER_SNAT))
+		src_sec_identity = WORLD_IPV4_ID;
+#else
+		if (src_sec_identity == 0)
+			src_sec_identity = WORLD_IPV4_ID;
+#endif
+
+		 /* Before forwarding the identity, make sure it's not a CIDR
+		  * identity, as these are __u32 values, but transporting them
+		  * via the VNI field in the VXLAN / Geneve header allows for
+		  * only 24 bits.
+		  */
```

Comment:
yeah, I think I'm just quoting the implementation detail (CIDR identities aka local identities don't fit in the VNI field) rather than stating clearly that forwarding local identities doesn't make sense, but that's the base idea